test/library/cluster_operator.go

```diff
@@ -3,38 +3,45 @@ package library
 import (
 	"fmt"
 	"testing"
+	"time"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	configv1 "github.com/openshift/api/config/v1"
-	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 	clusteroperatorhelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 )
 
 // WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded waits for ClusterOperator/kube-apiserver to report
 // status as active, not progressing, and not failing.
-func WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t *testing.T, client configclient.ConfigV1Interface) {
-	err := wait.Poll(WaitPollInterval, WaitPollTimeout, func() (bool, error) {
+func WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t *testing.T, client configv1client.ConfigV1Interface) {
+	t.Helper()
+	time.Sleep(time.Minute) // make sure we are not racing against an initial observation of change
```

Comment:
Right.  Consider the following scenario where you can race against KAS-O status updates:

1. KAS-O is idle at A=T, P=F, D=F
1. You make some change
1. KAS-O begins reacting to the change
1. You check if KAS-O is A=T, P=F, D=F
1. You see that KAS-O is A=T, P=F, D=F
1. You continue with the rest of the test
1. KAS-O updates its status A=T, P=T, D=F
1. Your test fails because it ran too early

Reply:
`sleeping` doesn't seem to be deterministic, relying on `kubeapiserver/cluster` resource as well. Did you consider waiting for a key to be migrated ?

Reply:
> Did you consider waiting for a key to be migrated ?

Waiting for _what_ key though?  Like you need to `sleep` anyway just to make sure that you see any "new" keys.  There is no guarantee that there is a _new_ key (or even old key) since current mode and new mode may be the same (since we randomize the order of tests).

In general the operator does not expose its internal state, on purpose.  There are preexisting tests that sleep for `5` seconds because they face similar issues.  If our control loops do not wake up in a minute, we have some other issue at hand.

Now I am experimenting with key based detection because I want to try to run the tests as fast as possible.  But I consider the current approach to be "good enough."